ow_plexil/src/plans/FaultHandlingPattern4.plp

```diff
@@ -0,0 +1,100 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+// If no faults are occurring, this plan will pan 180 degrees, 
+// and then unstow and stow the arm.
+// If any fault is injected, the plan will pause and wait for resolution.
+// If a fault is injected during an operation, that operation will only
+// halt immediately if the fault injected is associated with it directly,
+// i.e. an antenna fault during panning or an arm fault during 
+// unstowing or stowing.
+// Otherwise, the pausing will visually become apparent when the next
+// operation does not begin due to it.
+
+#include "plan-interface.h"
+
+LibraryAction InitializeAntennaAndArm ();
+
+FaultHandlingPattern4:
+{
+
+  Boolean ImageDone = false;
+  Boolean UnstowDone = false;
+  Boolean StowDone = false;
+
+  LibraryCall InitializeAntennaAndArm ();
+
+  log_info ("Starting FaultHandlingPattern4 plan...");
+
+  Mission: 
+  {
+    
+    // This plan current will restart nodes if they are interrupted by a fault
+    // and then that fault is resolved. This will continue until completion of
+    // the processes in that node. However, currently the plan will continue
+    // indefinitely on Image since there is no way to get the action state
+    // in PLEXIL. This is what Jira 778 is for, so this plan can be tested
+    // once that Jira has been resolved.
+
+    Image:
+    {
+      Repeat !ImageDone;
+      Start !Lookup(AntennaFault);
+      Exit ImageDone;
+
+      SynchronousCommand pan_antenna (180);
+
+      /*
+      if pan_antenna.state == SUCCEEDED // #TODO Fix syntax after jira 778 done
+      {
+        log_info ("Image successfully completed...");
+        ImageDone = true;
+      }
+      */
```

Comment:
Jira 778 won't fully solve the problem here, and is unlikely to be done soon.    Have you, or can you, try the following?

```
ImageDone =  ! Lookup(AntennaFault);
```
combined with : `Exit ImageDone || Lookup(AntennaFault);`

It's not perfect, but may cover many use cases.    The same approach applies to all the commented sections.

Reply:
Does this translate to Exit !Lookup(AntennaFault) || Lookup(AntennaFault)? Wouldn't this make it always exit? I might be missing something.

Reply:
No, as written it means exit if the image is done, or if there's an antenna fault.    I just realized that the former condition is not needed, since the node will end normally after the pan.   So the Exit could just be `Lookup(AntennaFault)`.

Note that my suggestion also implies that if you get past the SynchronousCommand and there's no antenna fault, you can consider ImageDone.  This is a gross simplification, but may cover many use cases.